File: backend/arguments-backend/src/main/java/com/asier/arguments/argumentsbackend/entities/DiscussionThread.java
package com.asier.arguments.argumentsbackend.entities;

import com.asier.arguments.argumentsbackend.misc.Identify;
import com.asier.arguments.argumentsbackend.utils.BasicUtils;
import com.asier.arguments.argumentsbackend.utils.annotations.Mandatory;
import com.asier.arguments.argumentsbackend.utils.annotations.Modifiable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZonedDateTime;
import java.util.HashSet;
import java.util.List;

/**
 * This class represents every discussion thread created by Users
 */
@Data
@Document(collection = "discussions")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DiscussionThread implements Identify {
    @Id
    private ObjectId id;
    @Mandatory
    private String title;
    private String author;
    private Instant createdAt;
    @Modifiable
    private Instant endAt;
    @Mandatory
    @Modifiable
    private Integer maxUsers;
    @Modifiable
    private HashSet<String> users;
    private DiscussionStatus status;

    @Override
    public String getId() {
        return BasicUtils.getIdentity(id);
    }
}

File: android/app/src/main/java/com/asier/arguments/entities/DiscussionThread.kt
package com.asier.arguments.entities

import java.time.Instant


data class DiscussionThread(
    var id: String = "",
    var title: String = "",
    var author: String = "",
    var maxUsers: Int = 0,
    var users: List<String> = emptyList(),
    var createdAt: Instant? = null,
    var endAt: Instant? = null,
    var votingGraceAt: Instant? = null,
    var votes: HashMap<String,Int> = hashMapOf(),
    var paimonVote: String = "",
    var status: DiscussionStatus = DiscussionStatus.NONE
)